### DISCUSSION TITLE
Laravel Bus::chain and Bus::batch Discrepancy?

### DISCUSSION THREAD

**Discussion by ahoiroman** [laravel/framework#49019] [Q&A]
Hello everyone,

I am currently working on a Laravel project utilizing the ﻿`Bus` facade's functionality to handle and organize my jobs. I noticed something peculiar about the implementation of ﻿`Bus::chain()` and ﻿`Bus::batch()`.
Specifically, it's the case that ﻿`Bus::chain()` does not provide a `﻿then()` method, while ﻿`Bus::batch()` does. Here's a basic overview of the usage:

## Bus::chain:

```
Bus::chain([
    new Job1(),
    new Job2(),
    //...
])->dispatch();
```

## Bus::batch:

```
Bus::batch([
    new Job1(),
    new Job2(),
    //...
])->then(function (Batch $batch) {
    // All jobs completed successfully...
})->dispatch();
```

Seeing as they're both methods to handle job execution, it's somewhat inconsistent. I would expect both methods to either have or not have the ﻿`then()` functionality.
Can anyone shed light on why this is the case? Why does `﻿Bus::chain()` not provide a `﻿then()` method but ﻿`Bus::batch()` does? Is there a specific design decision behind this? Any insights would be greatly appreciated.

**Comment by bretto36** [comment]
When it's batched there is a DB record that contains the then/catch/finally method calls. So i would assume that's the reason. There is no then on the Chain method cause there is no where to store it neatly?

If you want the jobs in the batch to run as a chain pass them in an array

```
Bus::batch([
    new Job1(),
    new Job2(),
    //...
])->then(function (Batch $batch) {
    // All jobs completed successfully...
})->dispatch();
```

becomes

```
Bus::batch([
    [
        new Job1(),
        new Job2(),
    ]
])->then(function (Batch $batch) {
    // All jobs completed successfully...
})->dispatch();
```
